assets/training/model_management/src/azureml/model/mgmt/utils/common_utils.py

```diff
@@ -180,44 +184,29 @@ def get_list_from_comma_separated_str(list_str: str, item_sep: str) -> List:
     return [x.strip() for x in list_str.split(item_sep) if x]
 
 
-def init_tc():
-    """Initialize app insights telemetry client."""
-    global tc
-    if tc is None:
-        try:
-            tc = TelemetryClient("71b954a8-6b7d-43f5-986c-3d3a6605d803")
-        except Exception as e:
-            print(f"Exception while initializing app insights: {e}")
-
-
-def tc_log(message):
-    """Log message to app insights."""
-    global tc
-    try:
-        print(message)
-        tc.track_event(name="FM_import_pipeline_debug_logs", properties={"message": message})
-        tc.flush()
-    except Exception as e:
-        print(f"Exception while logging to app insights: {e}")
-
-
-def tc_exception(e, message):
-    """Log exception to app insights."""
-    global tc
-    try:
-        tc.track_exception(value=e.__class__, properties={"exception": message})
-        tc.flush()
-    except Exception as e:
-        print(f"Exception while logging exception to app insights: {e}")
+def retry(times):
+    """Retry Decorator.
 
+    :param times: The number of times to repeat the wrapped function/method
+    :type times: Int
+    """
 
-def check_model_id(model_id):
-    """Hugging face model info."""
-    try:
-        model_list: List[ModelInfo] = HfApi(endpoint=HF_ENDPOINT).list_models(filter=ModelFilter(model_name=model_id))
-        for info in model_list:
-            if model_id == info.modelId:
-                return True
-    except Exception as e:
-        raise ValueError(f"Failed to validate model id : {e}")
-    return False
+    def decorator(func):
```

Comment:
Where is back off being defined before we retry the API call?

Reply:
if there is no exception, we would not be doing retry. Please let me know if that is what is being asked.

Reply:
do you mean delay before next retry? Havent defined as of now

Reply:
Yes. The comment was about exponential backoff in case of failure/exception.